### PR TITLE
IA-1645 Add checkbox to filter users by ou children or parent 

### PR DESCRIPTION
**Pull request by SimonDus**
On the profile API add of 2 checkboxes to filter users by Ou parent or children

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are there enough tests

## How to test

Create users with OU set on their profiles. 
Set OU Parent/Children on users and play with the filters. 
Selection an ou to do the search is required...

## Print screen / video

https://user-images.githubusercontent.com/45785235/209309416-3db1a05f-2976-4290-b8c1-4ed177d4a8d7.mov

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/users/components/Filters.js [hunk 1]
```diff
@@ -29,13 +29,16 @@ const useStyles = makeStyles(theme => ({
...
+    const [ouParent, setOuParent] = useState(false);
+    const [ouChildren, setOuChildren] = useState(false);
     const { formatMessage } = useSafeIntl();
     const dispatch = useDispatch();
     const [filters, setFilters] = useState({
         search: params.search,
         permissions: params.permissions,
         location: params.location,
         orgUnitTypes: params.orgUnitTypes,
+        ouParent: params.ouParent,
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/components/Filters.js:44]: shouldn't there be ouChildren here as well?

File: hat/assets/js/apps/Iaso/domains/users/messages.js [hunk 2]
```diff
@@ -217,6 +217,14 @@ const MESSAGES = defineMessages({
         defaultMessage: 'Home page',
         id: 'iaso.users.label.homePage',
     },
+    ouChildrenCheckbox: {
+        defaultMessage: 'Show OU Children',
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/messages.js]: Show org unit children

File: hat/assets/js/apps/Iaso/domains/users/messages.js [hunk 3]
```diff
@@ -217,6 +217,14 @@ const MESSAGES = defineMessages({
...
+        id: 'iaso.users.ouChildrenCheckbox',
+    },
+    ouParentCheckbox: {
+        defaultMessage: 'Show OU parent',
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/messages.js]: Show org unit parent

File: hat/assets/js/apps/Iaso/domains/users/components/Filters.js [hunk 4]
```diff
@@ -138,6 +149,36 @@ const Filters = ({ baseUrl, params }) => {
...
+                        type="checkbox"
+                        checked={ouParent}
+                        onChange={(key, value) => {
+                            handleChange('ouParent', value);
+                            setOuParent(value);
+                        }}
+                        disabled={ouChildren || !initialOrgUnit}
+                        value={ouParent}
+                        label={MESSAGES.ouParentCheckbox}
+                        // onEnterPressed={handleSearchPerms}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/components/Filters.js]: can be deleted

File: hat/assets/js/apps/Iaso/domains/users/components/Filters.js [hunk 5]
```diff
@@ -138,6 +149,36 @@ const Filters = ({ baseUrl, params }) => {
...
+                        type="checkbox"
+                        checked={ouChildren}
+                        onChange={(key, value) => {
+                            handleChange('ouChildren', !ouChildren);
+                            setOuChildren(value);
+                        }}
+                        disabled={ouParent || !initialOrgUnit}
+                        value={ouChildren}
+                        label={MESSAGES.ouChildrenCheckbox}
+                        // onEnterPressed={handleSearchPerms}
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/components/Filters.js]: can be deleted

File: hat/assets/js/apps/Iaso/domains/users/components/Filters.js [hunk 6]
```diff
@@ -26,16 +26,27 @@ const useStyles = makeStyles(theme => ({
     ...commonStyles(theme),
 }));
 
+const stringToBoolean = str => {
+    if (str === 'true') return true;
+    if (str === 'false') return false;
+    return undefined;
+};
+
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/components/Filters.js]: Can you move this to /iaso/hat/assets/apps/Iaso/utils/dataManipulation.ts?

**Review by quang-le — CHANGES_REQUESTED** [review]
- `ouChildren` is missing from the filter state, so it won't work when deep linking
- The checkboxes shouldn't be mutually exclusive

**Review by quang-le — CHANGES_REQUESTED** [review]
Thanks for the changes.

Is this the expected behaviour?

https://user-images.githubusercontent.com/38907762/209299964-f4689359-9ff8-44c7-a086-e8af23eee5f9.mov

Also:
-  The localisation (org unit) filter shouldn't push the checkboxes down when we select an org unit far down the tree
- We have (new) layout guidelines about filters. They shouldn't take the whole width of the screen. See for example Completeness stats:
Screenshot 2022-12-23 at 09 24 42
My take in this case is that we should have 3 filters in one row, and org units with the 2 checkboxes on another. @beygorghor  can you weight on this please?

**Review by quang-le — APPROVED** [review]
Nice job!

**Review by codeKameleon — APPROVED** [review]

**Review by beygorghor — APPROVED** [review]

**Comment by beygorghor** [comment]
I agree with @olethanh, for me it's not clear what those checkboxes are doing, maybe adding a info icon with a tooltip giving more infos can be a good idea

**Pull request merged**